src/Core/Module/ModuleManager.php

```diff
@@ -311,20 +312,35 @@ public function getError(string $name): string
         $module = $this->moduleRepository->getModule($name);
         if ($module->hasValidInstance()) {
             $errors = array_filter($module->getInstance()->getErrors());
-            $error = array_pop($errors);
-            if (empty($error)) {
+            if (empty($errors)) {
                 $error = $this->translator->trans(
                     'Unfortunately, the module %module% did not return additional details.',
                     ['%module%' => $name],
                     'Admin.Modules.Notification'
                 );
+            } else {
+                $error = implode(', ', $errors);
             }
         } else {
             $error = $this->translator->trans(
                 'The module %module% is invalid and cannot be loaded.',
                 ['%module%' => $name],
                 'Admin.Modules.Notification'
             );
+
+            $validityErrors = [];
+            if (!LegacyValidate::isModuleName($name)) {
+                $validityErrors[] = $name . ' module name is invalid';
+            }
+            try {
+                LegacyModule::getInstanceByName($name);
+            } catch (Throwable $e) {
+                $validityErrors[] = $e->getMessage();
+            }
```

Comment:
Hi @jolelievre 
Thanks, that's fine with me, just if the module name is invalid and dev mode is enabled, you'll get the "module name is invalid" message twice because `Module::getInstanceByName` already does that check. I think if the module name is invalid there's no point in trying to instantiate it.

Reply:
@PululuK, you're right. I put the second part in an `else` statement

By the way, the `die` in the `getInstanceByName` method is a bit scary, but not in the scope of this PR to improve it, and I worry and the impacts if try to replace it with an exception 😅